community/kernel/src/main/java/org/neo4j/GraphDatabaseBuilder.java

```diff
@@ -0,0 +1,384 @@
+            params.put( GraphDatabaseSettings.relationship_keys_indexable.name(), "" );
+        }
+        return self();
+    }
+
+    /**
+     * Disable the background index sampling.
+     *
+     * @return this builder
+     */
+    public BUILDER disableBackgroundIndexSampling()
+    {
+        params.put( GraphDatabaseSettings.index_background_sampling_enabled.name(), String.valueOf( false ) );
+        return self();
+    }
+
+    /**
+     * Set the size of buffer used by index sampling.
+     *
+     * @param bytes the size of buffer used by index sampling
+     * @return this builder
+     */
+    public BUILDER withIndexSamplingBufferSize( long bytes )
+    {
+        params.put( GraphDatabaseSettings.index_sampling_buffer_size.name(), String.valueOf( bytes ) );
+        return self();
+    }
+
+    /**
+     * Set the percentage of index updates of total index size required before sampling of a given index is triggered.
+     *
+     * @param percentage the percentage of index updates of total index size required before sampling of a given index
+     * is triggered
+     * @return this builder
+     */
+    public BUILDER withIndexSamplingUpdatePercentage( int percentage )
+    {
+        params.put( GraphDatabaseSettings.index_sampling_update_percentage.name(), String.valueOf( percentage ) );
+        return self();
+    }
+
+    /**
+     * Set the maximum number of open Lucene index searchers.
+     *
+     * @param count the maximum number of open Lucene index searchers
+     * @return this builder
+     */
+    public BUILDER withLuceneIndexSearcherCacheSize( int count )
+    {
+        params.put( GraphDatabaseSettings.lucene_searcher_cache_size.name(), String.valueOf( count ) );
+        return self();
+    }
+
+    /**
+     * Set the threshold, in days, after which logical logs will be pruned.
+     *
+     * @param thresholdDays the threshold, in days, after which logical logs will be pruned
+     * @return this builder
+     */
+    public BUILDER withLogicalLogPruningThresholdDays( int thresholdDays )
+    {
+        params.put( GraphDatabaseSettings.keep_logical_logs.name(), String.valueOf( thresholdDays ) + " days" );
+        return self();
+    }
+
+    /**
+     * Set the threshold, in bytes, after which logical logs will be pruned.
+     *
+     * @param thresholdSize the threshold, in bytes, after which logical logs will be pruned
+     * @return this builder
+     */
+    public BUILDER withLogicalLogPruningThresholdSize( long thresholdSize )
+    {
+        params.put( GraphDatabaseSettings.keep_logical_logs.name(), String.valueOf( thresholdSize / 1024 ) + " k" );
+        return self();
+    }
+
+    /**
+     * Set the maximum number of archived logical logs.
+     *
+     * @param count the maximum number of archived logical logs
+     * @return this builder
+     */
+    public BUILDER withLogicalLogMaxArchives( int count )
```

Comment:
This name is inconsistent with the other `LogicalLogPruning....` methods. Consistency would be great.